src/FastState.cpp

```diff
@@ -70,14 +70,14 @@ bool FastState::is_move_legal(int color, int vertex) {
                 !board.is_suicide(vertex, color));
 }
 
-void FastState::play_pass(void) {
+void FastState::play_pass(int color) {
     m_movenum++;
 
     std::rotate(rbegin(m_lastmove), rbegin(m_lastmove) + 1, rend(m_lastmove));
     m_lastmove[0] = FastBoard::PASS;
 
     board.m_hash  ^= 0xABCDABCDABCDABCDULL;
```

Comment:
`0xABCDABCDABCDABCDULL` looks like the marking for black's turn in the hash. 
If we repeatedly pass with the same color (possible in the leela interface), the hash would 
be wrong. So does this xoring also depend on the color that is passing?

Reply:
I think this can be fix by change that line to

```c++
    if (board.m_tomove!=color) board.m_hash  ^= 0xABCDABCDABCDABCDULL;
```

Reply:
This won't achieve anything because the hash includes the number of passes.